pkg/cmd/server/origin/master.go

```diff
@@ -782,6 +782,40 @@ func (c *MasterConfig) RunBuildController() {
 
 	controller := factory.Create()
 	controller.Run()
+	deleteController := factory.CreateDeleteController()
+	deleteController.Run()
+}
+
+// RunBuildController starts the build sync loop for builds and buildConfig processing.
+func (c *MasterConfig) RunBuildDeleteController() {
+	// initialize build controller
+	dockerImage := c.ImageFor("docker-builder")
+	stiImage := c.ImageFor("sti-builder")
+
+	osclient, kclient := c.BuildControllerClients()
+	factory := buildcontrollerfactory.BuildControllerFactory{
+		OSClient:     osclient,
+		KubeClient:   kclient,
+		BuildUpdater: buildclient.NewOSClientBuildClient(osclient),
+		DockerBuildStrategy: &buildstrategy.DockerBuildStrategy{
+			Image: dockerImage,
+			// TODO: this will be set to --storage-version (the internal schema we use)
+			Codec: v1beta1.Codec,
```

Comment:
Ugh, this todo needs to be resolved before we ship.  I need to spawn an issue on it.